packages/components/pie-avatar/src/index.ts

```diff
@@ -1,22 +1,105 @@
-import { html, unsafeCSS } from 'lit';
+import {
+    html, type TemplateResult, unsafeCSS,
+} from 'lit';
 import { PieElement } from '@justeattakeaway/pie-webc-core/src/internals/PieElement';
-import { RtlMixin, safeCustomElement } from '@justeattakeaway/pie-webc-core';
+import { RtlMixin, safeCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
 
+import { property } from 'lit/decorators.js';
 import styles from './avatar.scss?inline';
-import { type AvatarProps } from './defs';
+import {
+    type AvatarProps, defaultProps, tags,
+} from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-avatar';
 
+type Initials = {
+    visual: string,
+    screenreader: string
+}
+
 /**
  * @tagname pie-avatar
  */
 @safeCustomElement('pie-avatar')
 export class PieAvatar extends RtlMixin(PieElement) implements AvatarProps {
+    @property({ type: String })
+    @validPropertyValues(componentSelector, tags, defaultProps.tag)
+    public tag = defaultProps.tag; // access modifier
```

Comment:
i think we can omit the comment 
```suggestion
    public tag = defaultProps.tag;
```